{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Biobase\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, as.vector, cbind, colnames,\n",
      "    do.call, duplicated, eval, evalq, Filter, Find, get, grep, grepl,\n",
      "    intersect, is.unsorted, lapply, lengths, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unlist, unsplit\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Setting options('download.file.method.GEOquery'='auto')\n",
      "Setting options('GEOquery.inmemory.gpl'=FALSE)\n"
     ]
    }
   ],
   "source": [
    "library(GEOquery)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using locally cached version of GDS4102 found here:\n",
      "/scratch/RtmpSiIMGg/GDS4102.soft.gz \n"
     ]
    }
   ],
   "source": [
    "gds <- getGEO(\"GDS4102\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<dl>\n",
       "\t<dt>$channel_count</dt>\n",
       "\t\t<dd>'1'</dd>\n",
       "\t<dt>$dataset_id</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'GDS4102'</li>\n",
       "\t<li>'GDS4102'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$description</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'Analysis of tumor tissue and normal tissue in pancreatic cancer samples. The fresh frozen samples were obtained during surgical procedures. Results provide insight into molecular mechanisms underlying tumorigenesis.'</li>\n",
       "\t<li>'tumor'</li>\n",
       "\t<li>'normal'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$email</dt>\n",
       "\t\t<dd>'geo@ncbi.nlm.nih.gov'</dd>\n",
       "\t<dt>$feature_count</dt>\n",
       "\t\t<dd>'54675'</dd>\n",
       "\t<dt>$institute</dt>\n",
       "\t\t<dd>'NCBI NLM NIH'</dd>\n",
       "\t<dt>$name</dt>\n",
       "\t\t<dd>'Gene Expression Omnibus (GEO)'</dd>\n",
       "\t<dt>$order</dt>\n",
       "\t\t<dd>'none'</dd>\n",
       "\t<dt>$platform</dt>\n",
       "\t\t<dd>'GPL570'</dd>\n",
       "\t<dt>$platform_organism</dt>\n",
       "\t\t<dd>'Homo sapiens'</dd>\n",
       "\t<dt>$platform_technology_type</dt>\n",
       "\t\t<dd>'in situ oligonucleotide'</dd>\n",
       "\t<dt>$pubmed_id</dt>\n",
       "\t\t<dd>'19732725'</dd>\n",
       "\t<dt>$ref</dt>\n",
       "\t\t<dd>'Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6'</dd>\n",
       "\t<dt>$reference_series</dt>\n",
       "\t\t<dd>'GSE16515'</dd>\n",
       "\t<dt>$sample_count</dt>\n",
       "\t\t<dd>'52'</dd>\n",
       "\t<dt>$sample_id</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'GSM414924,GSM414925,GSM414926,GSM414927,GSM414929,GSM414931,GSM414933,GSM414935,GSM414936,GSM414937,GSM414939,GSM414941,GSM414943,GSM414944,GSM414945,GSM414946,GSM414948,GSM414949,GSM414950,GSM414951,GSM414952,GSM414954,GSM414956,GSM414958,GSM414959,GSM414960,GSM414961,GSM414962,GSM414964,GSM414965,GSM414967,GSM414968,GSM414969,GSM414971,GSM414973,GSM414974'</li>\n",
       "\t<li>'GSM414928,GSM414930,GSM414932,GSM414934,GSM414938,GSM414940,GSM414942,GSM414947,GSM414953,GSM414955,GSM414957,GSM414963,GSM414966,GSM414970,GSM414972,GSM414975'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$sample_organism</dt>\n",
       "\t\t<dd>'Homo sapiens'</dd>\n",
       "\t<dt>$sample_type</dt>\n",
       "\t\t<dd>'RNA'</dd>\n",
       "\t<dt>$title</dt>\n",
       "\t\t<dd>'Pancreatic Tumor and Normal tissue samples'</dd>\n",
       "\t<dt>$type</dt>\n",
       "\t\t<dd><ol class=list-inline>\n",
       "\t<li>'Expression profiling by array'</li>\n",
       "\t<li>'tissue'</li>\n",
       "\t<li>'tissue'</li>\n",
       "</ol>\n",
       "</dd>\n",
       "\t<dt>$update_date</dt>\n",
       "\t\t<dd>'Jun 07 2012'</dd>\n",
       "\t<dt>$value_type</dt>\n",
       "\t\t<dd>'count'</dd>\n",
       "\t<dt>$web_link</dt>\n",
       "\t\t<dd>'http://www.ncbi.nlm.nih.gov/geo'</dd>\n",
       "</dl>\n"
      ],
      "text/latex": [
       "\\begin{description}\n",
       "\\item[\\$channel_count] '1'\n",
       "\\item[\\$dataset_id] \\begin{enumerate*}\n",
       "\\item 'GDS4102'\n",
       "\\item 'GDS4102'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$description] \\begin{enumerate*}\n",
       "\\item 'Analysis of tumor tissue and normal tissue in pancreatic cancer samples. The fresh frozen samples were obtained during surgical procedures. Results provide insight into molecular mechanisms underlying tumorigenesis.'\n",
       "\\item 'tumor'\n",
       "\\item 'normal'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$email] 'geo@ncbi.nlm.nih.gov'\n",
       "\\item[\\$feature_count] '54675'\n",
       "\\item[\\$institute] 'NCBI NLM NIH'\n",
       "\\item[\\$name] 'Gene Expression Omnibus (GEO)'\n",
       "\\item[\\$order] 'none'\n",
       "\\item[\\$platform] 'GPL570'\n",
       "\\item[\\$platform_organism] 'Homo sapiens'\n",
       "\\item[\\$platform_technology_type] 'in situ oligonucleotide'\n",
       "\\item[\\$pubmed_id] '19732725'\n",
       "\\item[\\$ref] 'Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6'\n",
       "\\item[\\$reference_series] 'GSE16515'\n",
       "\\item[\\$sample_count] '52'\n",
       "\\item[\\$sample_id] \\begin{enumerate*}\n",
       "\\item 'GSM414924,GSM414925,GSM414926,GSM414927,GSM414929,GSM414931,GSM414933,GSM414935,GSM414936,GSM414937,GSM414939,GSM414941,GSM414943,GSM414944,GSM414945,GSM414946,GSM414948,GSM414949,GSM414950,GSM414951,GSM414952,GSM414954,GSM414956,GSM414958,GSM414959,GSM414960,GSM414961,GSM414962,GSM414964,GSM414965,GSM414967,GSM414968,GSM414969,GSM414971,GSM414973,GSM414974'\n",
       "\\item 'GSM414928,GSM414930,GSM414932,GSM414934,GSM414938,GSM414940,GSM414942,GSM414947,GSM414953,GSM414955,GSM414957,GSM414963,GSM414966,GSM414970,GSM414972,GSM414975'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$sample_organism] 'Homo sapiens'\n",
       "\\item[\\$sample_type] 'RNA'\n",
       "\\item[\\$title] 'Pancreatic Tumor and Normal tissue samples'\n",
       "\\item[\\$type] \\begin{enumerate*}\n",
       "\\item 'Expression profiling by array'\n",
       "\\item 'tissue'\n",
       "\\item 'tissue'\n",
       "\\end{enumerate*}\n",
       "\n",
       "\\item[\\$update_date] 'Jun 07 2012'\n",
       "\\item[\\$value_type] 'count'\n",
       "\\item[\\$web_link] 'http://www.ncbi.nlm.nih.gov/geo'\n",
       "\\end{description}\n"
      ],
      "text/markdown": [
       "$channel_count\n",
       ":   '1'\n",
       "$dataset_id\n",
       ":   1. 'GDS4102'\n",
       "2. 'GDS4102'\n",
       "\n",
       "\n",
       "\n",
       "$description\n",
       ":   1. 'Analysis of tumor tissue and normal tissue in pancreatic cancer samples. The fresh frozen samples were obtained during surgical procedures. Results provide insight into molecular mechanisms underlying tumorigenesis.'\n",
       "2. 'tumor'\n",
       "3. 'normal'\n",
       "\n",
       "\n",
       "\n",
       "$email\n",
       ":   'geo@ncbi.nlm.nih.gov'\n",
       "$feature_count\n",
       ":   '54675'\n",
       "$institute\n",
       ":   'NCBI NLM NIH'\n",
       "$name\n",
       ":   'Gene Expression Omnibus (GEO)'\n",
       "$order\n",
       ":   'none'\n",
       "$platform\n",
       ":   'GPL570'\n",
       "$platform_organism\n",
       ":   'Homo sapiens'\n",
       "$platform_technology_type\n",
       ":   'in situ oligonucleotide'\n",
       "$pubmed_id\n",
       ":   '19732725'\n",
       "$ref\n",
       ":   'Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6'\n",
       "$reference_series\n",
       ":   'GSE16515'\n",
       "$sample_count\n",
       ":   '52'\n",
       "$sample_id\n",
       ":   1. 'GSM414924,GSM414925,GSM414926,GSM414927,GSM414929,GSM414931,GSM414933,GSM414935,GSM414936,GSM414937,GSM414939,GSM414941,GSM414943,GSM414944,GSM414945,GSM414946,GSM414948,GSM414949,GSM414950,GSM414951,GSM414952,GSM414954,GSM414956,GSM414958,GSM414959,GSM414960,GSM414961,GSM414962,GSM414964,GSM414965,GSM414967,GSM414968,GSM414969,GSM414971,GSM414973,GSM414974'\n",
       "2. 'GSM414928,GSM414930,GSM414932,GSM414934,GSM414938,GSM414940,GSM414942,GSM414947,GSM414953,GSM414955,GSM414957,GSM414963,GSM414966,GSM414970,GSM414972,GSM414975'\n",
       "\n",
       "\n",
       "\n",
       "$sample_organism\n",
       ":   'Homo sapiens'\n",
       "$sample_type\n",
       ":   'RNA'\n",
       "$title\n",
       ":   'Pancreatic Tumor and Normal tissue samples'\n",
       "$type\n",
       ":   1. 'Expression profiling by array'\n",
       "2. 'tissue'\n",
       "3. 'tissue'\n",
       "\n",
       "\n",
       "\n",
       "$update_date\n",
       ":   'Jun 07 2012'\n",
       "$value_type\n",
       ":   'count'\n",
       "$web_link\n",
       ":   'http://www.ncbi.nlm.nih.gov/geo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "$channel_count\n",
       "[1] \"1\"\n",
       "\n",
       "$dataset_id\n",
       "[1] \"GDS4102\" \"GDS4102\"\n",
       "\n",
       "$description\n",
       "[1] \"Analysis of tumor tissue and normal tissue in pancreatic cancer samples. The fresh frozen samples were obtained during surgical procedures. Results provide insight into molecular mechanisms underlying tumorigenesis.\"\n",
       "[2] \"tumor\"                                                                                                                                                                                                                  \n",
       "[3] \"normal\"                                                                                                                                                                                                                 \n",
       "\n",
       "$email\n",
       "[1] \"geo@ncbi.nlm.nih.gov\"\n",
       "\n",
       "$feature_count\n",
       "[1] \"54675\"\n",
       "\n",
       "$institute\n",
       "[1] \"NCBI NLM NIH\"\n",
       "\n",
       "$name\n",
       "[1] \"Gene Expression Omnibus (GEO)\"\n",
       "\n",
       "$order\n",
       "[1] \"none\"\n",
       "\n",
       "$platform\n",
       "[1] \"GPL570\"\n",
       "\n",
       "$platform_organism\n",
       "[1] \"Homo sapiens\"\n",
       "\n",
       "$platform_technology_type\n",
       "[1] \"in situ oligonucleotide\"\n",
       "\n",
       "$pubmed_id\n",
       "[1] \"19732725\"\n",
       "\n",
       "$ref\n",
       "[1] \"Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6\"\n",
       "\n",
       "$reference_series\n",
       "[1] \"GSE16515\"\n",
       "\n",
       "$sample_count\n",
       "[1] \"52\"\n",
       "\n",
       "$sample_id\n",
       "[1] \"GSM414924,GSM414925,GSM414926,GSM414927,GSM414929,GSM414931,GSM414933,GSM414935,GSM414936,GSM414937,GSM414939,GSM414941,GSM414943,GSM414944,GSM414945,GSM414946,GSM414948,GSM414949,GSM414950,GSM414951,GSM414952,GSM414954,GSM414956,GSM414958,GSM414959,GSM414960,GSM414961,GSM414962,GSM414964,GSM414965,GSM414967,GSM414968,GSM414969,GSM414971,GSM414973,GSM414974\"\n",
       "[2] \"GSM414928,GSM414930,GSM414932,GSM414934,GSM414938,GSM414940,GSM414942,GSM414947,GSM414953,GSM414955,GSM414957,GSM414963,GSM414966,GSM414970,GSM414972,GSM414975\"                                                                                                                                                                                                        \n",
       "\n",
       "$sample_organism\n",
       "[1] \"Homo sapiens\"\n",
       "\n",
       "$sample_type\n",
       "[1] \"RNA\"\n",
       "\n",
       "$title\n",
       "[1] \"Pancreatic Tumor and Normal tissue samples\"\n",
       "\n",
       "$type\n",
       "[1] \"Expression profiling by array\" \"tissue\"                       \n",
       "[3] \"tissue\"                       \n",
       "\n",
       "$update_date\n",
       "[1] \"Jun 07 2012\"\n",
       "\n",
       "$value_type\n",
       "[1] \"count\"\n",
       "\n",
       "$web_link\n",
       "[1] \"http://www.ncbi.nlm.nih.gov/geo\"\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Meta(gds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using locally cached version of GPL570 found here:\n",
      "/scratch/RtmpSiIMGg/GPL570.annot.gz \n",
      "Warning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In readLines(con, 1): seek on a gzfile connection returned an internal errorWarning message:\n",
      "In read.table(con, sep = sep, header = header, nrows = sampleRows, : seek on a gzfile connection returned an internal errorWarning message:\n",
      "In read.table(con, sep = sep, header = header, nrows = sampleRows, : seek on a gzfile connection returned an internal errorWarning message:\n",
      "In read.table(file = file, header = header, sep = sep, quote = quote, : seek on a gzfile connection returned an internal error"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ExpressionSet (storageMode: lockedEnvironment)\n",
       "assayData: 1 features, 52 samples \n",
       "  element names: exprs \n",
       "protocolData: none\n",
       "phenoData\n",
       "  sampleNames: GSM414924 GSM414925 ... GSM414975 (52 total)\n",
       "  varLabels: sample tissue description\n",
       "  varMetadata: labelDescription\n",
       "featureData\n",
       "  featureNames: 1007_s_at\n",
       "  fvarLabels: ID Gene title ... GO:Component ID (21 total)\n",
       "  fvarMetadata: Column labelDescription\n",
       "experimentData: use 'experimentData(object)'\n",
       "  pubMedIds: 19732725 \n",
       "Annotation:  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "eset <- GDS2eSet(gds,do.log2=TRUE, )\n",
    "head(eset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>sample</th><th scope=col>tissue</th><th scope=col>description</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GSM414924</th><td>GSM414924</td><td>tumor</td><td>Value for GSM414924: Pancreatic Sample 6-Tumor; src: PancreaticSample6-Tumor</td></tr>\n",
       "\t<tr><th scope=row>GSM414925</th><td>GSM414925</td><td>tumor</td><td>Value for GSM414925: Pancreatic Sample 7-Tumor; src: PancreaticSample7-Tumor</td></tr>\n",
       "\t<tr><th scope=row>GSM414926</th><td>GSM414926</td><td>tumor</td><td>Value for GSM414926: Pancreatic Sample 1-Tumor; src: PancreaticSample1-Tumor</td></tr>\n",
       "\t<tr><th scope=row>GSM414927</th><td>GSM414927</td><td>tumor</td><td>Value for GSM414927: Pancreatic Sample 16-Tumor; src: PancreaticSample16-Tumor</td></tr>\n",
       "\t<tr><th scope=row>GSM414929</th><td>GSM414929</td><td>tumor</td><td>Value for GSM414929: Pancreatic Sample 53-Tumor; src: PancreaticSample53-Tumor</td></tr>\n",
       "\t<tr><th scope=row>GSM414931</th><td>GSM414931</td><td>tumor</td><td>Value for GSM414931: Pancreatic Sample 9-Tumor; src: PancreaticSample9-Tumor</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       "  & sample & tissue & description\\\\\n",
       "\\hline\n",
       "\tGSM414924 & GSM414924 & tumor & Value for GSM414924: Pancreatic Sample 6-Tumor; src: PancreaticSample6-Tumor\\\\\n",
       "\tGSM414925 & GSM414925 & tumor & Value for GSM414925: Pancreatic Sample 7-Tumor; src: PancreaticSample7-Tumor\\\\\n",
       "\tGSM414926 & GSM414926 & tumor & Value for GSM414926: Pancreatic Sample 1-Tumor; src: PancreaticSample1-Tumor\\\\\n",
       "\tGSM414927 & GSM414927 & tumor & Value for GSM414927: Pancreatic Sample 16-Tumor; src: PancreaticSample16-Tumor\\\\\n",
       "\tGSM414929 & GSM414929 & tumor & Value for GSM414929: Pancreatic Sample 53-Tumor; src: PancreaticSample53-Tumor\\\\\n",
       "\tGSM414931 & GSM414931 & tumor & Value for GSM414931: Pancreatic Sample 9-Tumor; src: PancreaticSample9-Tumor\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/plain": [
       "             sample tissue\n",
       "GSM414924 GSM414924  tumor\n",
       "GSM414925 GSM414925  tumor\n",
       "GSM414926 GSM414926  tumor\n",
       "GSM414927 GSM414927  tumor\n",
       "GSM414929 GSM414929  tumor\n",
       "GSM414931 GSM414931  tumor\n",
       "                                                                             description\n",
       "GSM414924   Value for GSM414924: Pancreatic Sample 6-Tumor; src: PancreaticSample6-Tumor\n",
       "GSM414925   Value for GSM414925: Pancreatic Sample 7-Tumor; src: PancreaticSample7-Tumor\n",
       "GSM414926   Value for GSM414926: Pancreatic Sample 1-Tumor; src: PancreaticSample1-Tumor\n",
       "GSM414927 Value for GSM414927: Pancreatic Sample 16-Tumor; src: PancreaticSample16-Tumor\n",
       "GSM414929 Value for GSM414929: Pancreatic Sample 53-Tumor; src: PancreaticSample53-Tumor\n",
       "GSM414931   Value for GSM414931: Pancreatic Sample 9-Tumor; src: PancreaticSample9-Tumor"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(pData(eset))\n",
    "library(geoq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for Converting {GEOquery}\"><tr><td>Converting {GEOquery}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Convert a GDS data structure to a BioConductor data structure</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Functions to take a GDS data structure from getGEO and coerce it to\n",
       "limma MALists or ExpressionSets.  \n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "GDS2MA(GDS,do.log2=FALSE,GPL=NULL,AnnotGPL=TRUE,getGPL=TRUE)\n",
       "GDS2eSet(GDS,do.log2=FALSE,GPL=NULL,AnnotGPL=TRUE,getGPL=TRUE)\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>GDS</code></td>\n",
       "<td>\n",
       "<p>The GDS datastructure returned by getGEO</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>do.log2</code></td>\n",
       "<td>\n",
       "<p>Boolean, should the data in the GDS be log2 transformed\n",
       "before inserting into the new data structure</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>GPL</code></td>\n",
       "<td>\n",
       "<p>Either a GPL data structure (from a call to getGEO) or\n",
       "NULL.  If NULL, this will cause a call to getGEO to produce a GPL.\n",
       "The gene information from the GPL is then used to construct the\n",
       "<code>genes</code> slot of the resulting limma <code>MAList</code> object or the\n",
       "<code>featureData</code> slot of the <code>ExpressionSet</code> instance. </p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>AnnotGPL</code></td>\n",
       "<td>\n",
       "<p>In general, the annotation GPL files will be available\n",
       "for GDS records, so the default is to use these files over the\n",
       "user-submitted GPL files</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>getGPL</code></td>\n",
       "<td>\n",
       "<p>A boolean defaulting to TRUE as to whether or not to\n",
       "download and include GPL information when converting to\n",
       "ExpressionSet or MAList.\n",
       "You may want to set this to FALSE if you know that you are going to\n",
       "annotate your featureData using Bioconductor tools rather than relying\n",
       "on information provided through NCBI GEO.  Download times can also\n",
       "be greatly reduced by specifying FALSE.</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Details</h3>\n",
       "\n",
       "<p>This function just rearranges one data structure into another.  For\n",
       "GDS, it also deals appropriately with making the &quot;targets&quot; list item\n",
       "for the limma data structure and the phenoData slot of ExpressionSets.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Value</h3>\n",
       "\n",
       "<table summary=\"R valueblock\">\n",
       "<tr valign=\"top\"><td><code>GDS2MA</code></td>\n",
       "<td>\n",
       "<p>A limma MAList</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>GDS2eSet</code></td>\n",
       "<td>\n",
       "<p>An ExpressionSet object</p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Author(s)</h3>\n",
       "\n",
       "<p>Sean Davis</p>\n",
       "\n",
       "\n",
       "<h3>References</h3>\n",
       "\n",
       "<p>See the limma and ExpressionSet help in the appropriate packages</p>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "\n",
       "## Not run: gds505 &lt;- getGEO('GDS505')\n",
       "## Not run: MA &lt;- GDS2MA(gds505)\n",
       "## Not run: eset &lt;- GDS2eSet(gds505)\n",
       "\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>GEOquery</em> version 2.36.0 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{Converting}{Convert a GDS data structure to a BioConductor data structure}{Converting}\n",
       "\\aliasA{GDS2eSet}{Converting}{GDS2eSet}\n",
       "\\aliasA{GDS2MA}{Converting}{GDS2MA}\n",
       "\\keyword{IO}{Converting}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "Functions to take a GDS data structure from getGEO and coerce it to\n",
       "limma MALists or ExpressionSets.  \n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "GDS2MA(GDS,do.log2=FALSE,GPL=NULL,AnnotGPL=TRUE,getGPL=TRUE)\n",
       "GDS2eSet(GDS,do.log2=FALSE,GPL=NULL,AnnotGPL=TRUE,getGPL=TRUE)\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{GDS}] The GDS datastructure returned by getGEO\n",
       "\\item[\\code{do.log2}] Boolean, should the data in the GDS be log2 transformed\n",
       "before inserting into the new data structure\n",
       "\\item[\\code{GPL}] Either a GPL data structure (from a call to getGEO) or\n",
       "NULL.  If NULL, this will cause a call to getGEO to produce a GPL.\n",
       "The gene information from the GPL is then used to construct the\n",
       "\\code{genes} slot of the resulting limma \\code{MAList} object or the\n",
       "\\code{featureData} slot of the \\code{ExpressionSet} instance. \n",
       "\\item[\\code{AnnotGPL}] In general, the annotation GPL files will be available\n",
       "for GDS records, so the default is to use these files over the\n",
       "user-submitted GPL files\n",
       "\\item[\\code{getGPL}] A boolean defaulting to TRUE as to whether or not to\n",
       "download and include GPL information when converting to\n",
       "ExpressionSet or MAList.\n",
       "You may want to set this to FALSE if you know that you are going to\n",
       "annotate your featureData using Bioconductor tools rather than relying\n",
       "on information provided through NCBI GEO.  Download times can also\n",
       "be greatly reduced by specifying FALSE.\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Details}\\relax\n",
       "This function just rearranges one data structure into another.  For\n",
       "GDS, it also deals appropriately with making the \"targets\" list item\n",
       "for the limma data structure and the phenoData slot of ExpressionSets.\n",
       "\\end{Details}\n",
       "%\n",
       "\\begin{Value}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{GDS2MA}] A limma MAList\n",
       "\\item[\\code{GDS2eSet}] An ExpressionSet object\n",
       "\\end{ldescription}\n",
       "\\end{Value}\n",
       "%\n",
       "\\begin{Author}\\relax\n",
       "Sean Davis\n",
       "\\end{Author}\n",
       "%\n",
       "\\begin{References}\\relax\n",
       "See the limma and ExpressionSet help in the appropriate packages\n",
       "\\end{References}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "\n",
       "## Not run: gds505 <- getGEO('GDS505')\n",
       "## Not run: MA <- GDS2MA(gds505)\n",
       "## Not run: eset <- GDS2eSet(gds505)\n",
       "\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "Converting              package:GEOquery               R Documentation\n",
       "\n",
       "_\bC_\bo_\bn_\bv_\be_\br_\bt _\ba _\bG_\bD_\bS _\bd_\ba_\bt_\ba _\bs_\bt_\br_\bu_\bc_\bt_\bu_\br_\be _\bt_\bo _\ba _\bB_\bi_\bo_\bC_\bo_\bn_\bd_\bu_\bc_\bt_\bo_\br _\bd_\ba_\bt_\ba _\bs_\bt_\br_\bu_\bc_\bt_\bu_\br_\be\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Functions to take a GDS data structure from getGEO and coerce it\n",
       "     to limma MALists or ExpressionSets.\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     GDS2MA(GDS,do.log2=FALSE,GPL=NULL,AnnotGPL=TRUE,getGPL=TRUE)\n",
       "     GDS2eSet(GDS,do.log2=FALSE,GPL=NULL,AnnotGPL=TRUE,getGPL=TRUE)\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "     GDS: The GDS datastructure returned by getGEO\n",
       "\n",
       " do.log2: Boolean, should the data in the GDS be log2 transformed\n",
       "          before inserting into the new data structure\n",
       "\n",
       "     GPL: Either a GPL data structure (from a call to getGEO) or NULL.\n",
       "          If NULL, this will cause a call to getGEO to produce a GPL.\n",
       "          The gene information from the GPL is then used to construct\n",
       "          the ‘genes’ slot of the resulting limma ‘MAList’ object or\n",
       "          the ‘featureData’ slot of the ‘ExpressionSet’ instance.\n",
       "\n",
       "AnnotGPL: In general, the annotation GPL files will be available for\n",
       "          GDS records, so the default is to use these files over the\n",
       "          user-submitted GPL files\n",
       "\n",
       "  getGPL: A boolean defaulting to TRUE as to whether or not to download\n",
       "          and include GPL information when converting to ExpressionSet\n",
       "          or MAList.  You may want to set this to FALSE if you know\n",
       "          that you are going to annotate your featureData using\n",
       "          Bioconductor tools rather than relying on information\n",
       "          provided through NCBI GEO.  Download times can also be\n",
       "          greatly reduced by specifying FALSE.\n",
       "\n",
       "_\bD_\be_\bt_\ba_\bi_\bl_\bs:\n",
       "\n",
       "     This function just rearranges one data structure into another.\n",
       "     For GDS, it also deals appropriately with making the \"targets\"\n",
       "     list item for the limma data structure and the phenoData slot of\n",
       "     ExpressionSets.\n",
       "\n",
       "_\bV_\ba_\bl_\bu_\be:\n",
       "\n",
       "  GDS2MA: A limma MAList\n",
       "\n",
       "GDS2eSet: An ExpressionSet object\n",
       "\n",
       "_\bA_\bu_\bt_\bh_\bo_\br(_\bs):\n",
       "\n",
       "     Sean Davis\n",
       "\n",
       "_\bR_\be_\bf_\be_\br_\be_\bn_\bc_\be_\bs:\n",
       "\n",
       "     See the limma and ExpressionSet help in the appropriate packages\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     ## Not run: gds505 <- getGEO('GDS505')\n",
       "     ## Not run: MA <- GDS2MA(gds505)\n",
       "     ## Not run: eset <- GDS2eSet(gds505)\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?GDS2eSet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.2.3 (2015-12-10)\n",
       "Platform: x86_64-pc-linux-gnu (64-bit)\n",
       "Running under: CentOS release 6.7 (Final)\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] parallel  stats     graphics  grDevices utils     datasets  methods  \n",
       "[8] base     \n",
       "\n",
       "other attached packages:\n",
       "[1] GEOquery_2.36.0     Biobase_2.30.0      BiocGenerics_0.16.1\n",
       "[4] ggplot2_2.1.0       dplyr_0.4.3        \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Rcpp_0.12.3      magrittr_1.5     munsell_0.4.3    uuid_0.1-2      \n",
       " [5] colorspace_1.2-6 R6_2.1.2         stringr_1.0.0    plyr_1.8.3      \n",
       " [9] tools_3.2.3      grid_3.2.3       gtable_0.2.0     DBI_0.3.1       \n",
       "[13] assertthat_0.1   digest_0.6.9     IRdisplay_0.3    repr_0.4        \n",
       "[17] bitops_1.0-6     base64enc_0.1-3  RCurl_1.95-4.8   IRkernel_0.6    \n",
       "[21] evaluate_0.8.3   pbdZMQ_0.2-1     stringi_1.0-1    scales_0.4.0    \n",
       "[25] XML_3.98-1.4     jsonlite_0.9.19 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.2.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
